op-batcher/batcher/channel_builder.go

```diff
@@ -127,12 +134,12 @@ type channelBuilder struct {
 
 // newChannelBuilder creates a new channel builder or returns an error if the
 // channel out could not be created.
-func newChannelBuilder(cfg ChannelConfig) (*channelBuilder, error) {
+func newChannelBuilder(cfg ChannelConfig, spanBatchBuilder *derive.SpanBatchBuilder) (*channelBuilder, error) {
```

Comment:
Why add parameter `spanBatchBuilder`? `cfg` already has `BatchType` so `newChannelBuilder` method can initialize itself.

Reply:
Hmm then how about adding all the necessary fields to `ChannelConfig` struct, to initialize spanbatchbuilder when necessary?